buffer/src/test/java/io/netty/buffer/AbstractByteBufTest.java

```diff
@@ -4939,4 +4939,63 @@ public void testEnsureWritableIntegerOverflow() {
             buffer.release();
         }
     }
+
+    @Test
+    public void testEndiannessIndexOf() {
+        buffer.clear();
+        final int v = 0x02030201;
+        buffer.writeIntLE(v);
+        buffer.writeByte(0x01);
+
+        assertEquals(-1, buffer.indexOf(1, 4, (byte) 1));
+        assertEquals(-1, buffer.indexOf(4, 1, (byte) 1));
+        assertEquals(1, buffer.indexOf(1, 4, (byte) 2));
+        assertEquals(3, buffer.indexOf(4, 1, (byte) 2));
+    }
+
+    @Test
+    public void explicitLittleEndianReadMethodsMustAlwaysUseLittleEndianByteOrder() {
```

Comment:
Don't you think the method name is too long?

Reply:
fine as it is... its a test after all and the method name at least describes what it tests.

Reply:
Alright.

Reply:
It's good to give tests names that describe what's being verified. This helps long-term test maintainability, because a common problem there is figuring out what an old but newly failing test was intended to verify.